_integration-test/run.py

```diff
@@ -0,0 +1,197 @@
+import subprocess
+import os
+from functools import lru_cache
+from bs4 import BeautifulSoup
+import httpx
+import pytest
+import sentry_sdk
+import time
+import json
+import re
+
+SENTRY_CONFIG_PY = "sentry/sentry.conf.py"
+SENTRY_TEST_HOST = os.getenv("SENTRY_TEST_HOST", "http://localhost:9000")
+TEST_USER = "test@example.com"
+TEST_PASS = "test123TEST"
+TIMEOUT_SECONDS = 60
+
+
+def poll_for_response(request: str, client: httpx.Client) -> httpx.Response:
+    for i in range(TIMEOUT_SECONDS):
+        response = client.get(
+            request, follow_redirects=True, headers={"Referer": SENTRY_TEST_HOST}
+        )
+        if response.status_code == 200:
```

Comment:
This function won't work for spans as it's waiting for a span to be ingested to succeed, so it's receiving a 200 no matter what and looking for the `data` field to not be empty.

We risk flaky tests if we leave it at that. Maybe we can have a custom behavior via a function to return `True`/`False` if it's considered a success of if we need to retry?